src/sql-parser/src/parser.rs

```diff
@@ -1692,6 +1694,38 @@ impl Parser {
         }))
     }
 
+    fn parse_create_type(&mut self) -> Result<Statement, ParserError> {
+        let name = self.parse_identifier()?;
+        self.expect_keyword("AS")?;
+        self.expect_keyword("MAP")?;
+
+        self.expect_token(&Token::LParen)?;
+        let first = self.parse_sql_option()?;
+        self.expect_token(&Token::Comma)?;
+        let second = self.parse_sql_option()?;
+        self.expect_token(&Token::RParen)?;
+
+        let (key, value) = match (first.name().as_str(), second.name().as_str()) {
+            ("key_type", "value_type") => (first, second),
+            ("value_type", "key_type") => (second, first),
+            (first, second) => {
+                return Err(self.error(
+                    self.peek_prev_range(),
+                    format!(
+                        "Expected key_type and value_type parameters, found {} and {}",
+                        first, second
+                    ),
+                ))
+            }
+        };
```

Comment:
I think it should be possible to eventually use `parse_with_options()` here, but that's just a refactor now (should be functionally equivalent), so this seems good for now!